state/postgresql/v2/postgresql.go

```diff
@@ -63,36 +71,57 @@ func NewPostgreSQLStateStore(logger logger.Logger) state.Store {
 	return NewPostgreSQLStateStoreWithOptions(logger, Options{})
 }
 
+// NewPostgreSQLStateStoreWithTrueOptions creates a new instance of PostgreSQL state store v2 with Azure AD authentication and AWS IAM authentication disabled.
+// The v2 of the component uses a different format for storing data, always in a BYTEA column, which is more efficient than the JSONB column used in v1.
+// Additionally, v2 uses random UUIDs for etags instead of the xmin column, expanding support to all Postgres-compatible databases such as CockroachDB, etc.
+func NewPostgreSQLStateStoreWithTrueOptions(logger logger.Logger) state.Store {
+	return NewPostgreSQLStateStoreWithOptions(logger, Options{NoAWSIAM: true, NoAzureAD: true})
+}
+
 // NewPostgreSQLStateStoreWithOptions creates a new instance of PostgreSQL state store with options.
 func NewPostgreSQLStateStoreWithOptions(logger logger.Logger, opts Options) state.Store {
 	s := &PostgreSQL{
 		logger:        logger,
 		enableAzureAD: !opts.NoAzureAD,
+		enableAWSIAM:  !opts.NoAWSIAM,
 	}
 	s.BulkStore = state.NewDefaultBulkStore(s)
 	return s
 }
 
 // Init sets up Postgres connection and performs migrations
 func (p *PostgreSQL) Init(ctx context.Context, meta state.Metadata) error {
-	err := p.metadata.InitWithMetadata(meta, p.enableAzureAD)
+	var (
+		useAWS   bool
+		useAzure bool
+		err      error
+	)
+
+	awsIam, _ := metadata.GetMetadataProperty(meta.Properties, "UseAWSIAM")
+	useAWS = utils.IsTruthy(awsIam)
```

Comment:
There is no need for any of this - our custom DecodeMetadata function already performs the IsTruthy check via decodehooks.

Please do not use `metadata.GetMetadataProperty`.

Instead, make sure on the original component metadata you have a property `UseAWSIAM` of type bool (or *bool if you do not want a default value). Then only access the value via the metadata struct.

Reply:
doing this the same as the azure setup, see my comment above on this for environment settings:
https://github.com/dapr/components-contrib/pull/3324#discussion_r1563051803